name.go

```diff
@@ -16,73 +16,45 @@ limitations under the License.
 
 package logicalcluster
 
-import (
-	"encoding/json"
-	"path"
-	"regexp"
-	"strings"
-)
-
-// ClusterHeader set to "<lcluster>" on a request is an alternative to accessing the
-// cluster via /clusters/<lcluster>. With that the <lcluster> can be access via normal kube-like
-// /api and /apis endpoints.
-const ClusterHeader = "X-Kubernetes-Cluster"
-
-// Name is the name of a logical cluster. A logical cluster is
-// 1. a (part of) etcd prefix to store objects in that cluster
-// 2. a (part of) a http path which serves a Kubernetes-cluster-like API with
-//    discovery, OpenAPI and the actual API groups.
-// 3. a value in metadata.clusterName in objects from cross-workspace list/watches,
-//    which is used to identify the logical cluster.
-//
-// A logical cluster is a colon separated list of words. In other words, it is
-// like a path, but with colons instead of slashes.
-type Name struct {
-	value string
-}
-
-const separator = ":"
+import "regexp"
 
 var (
-	// Wildcard is the name indicating cross-workspace requests.
-	Wildcard = New("*")
-
-	// None is the name indicating a cluster-unaware context.
-	None = New("")
+	clusterNameRegExp = regexp.MustCompile(clusterNameString)
+)
 
-	// TODO is a value created by automated refactoring tools that should be replaced by a real Name.
-	TODO = None
+const (
+	clusterNameString string = "^[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$"
 )
 
-// New returns a Name from a string.
-func New(value string) Name {
-	return Name{value}
-}
+// Name represents a cluster name, which:
+//  1. can be used to access a cluster via `/cluster/<cluster-name>`
+//  2. is part of an etcd key path
+//  3. is used to uniquely reference a logical cluster.
+//     there is at most one cluster name for a logical cluster,
+//     but many logical cluster identifiers can alias one cluster name.
+type Name string
 
-// NewValidated returns a Name from a string and whether it is a valid logical cluster.
-// A valid logical cluster returns true on IsValid().
-func NewValidated(value string) (Name, bool) {
-	n := Name{value}
-	return n, n.IsValid()
+// Path returns a Path (a logical cluster) out of the Name.
+func (n Name) Path() Path {
+	return New(string(n))
 }
 
-// Empty returns true if the logical cluster value is unset.
-func (n Name) Empty() bool {
-	return n.value == ""
+// String returns the string representation of the cluster name.
+func (n Name) String() string {
+	return string(n)
 }
 
-// Path returns a path segment for the logical cluster to access its API.
-func (n Name) Path() string {
-	return path.Join("/clusters", n.value)
+// IsValid returns true if the name starts with a lower-case letter and contains only lower-case letters, digits and hyphens.
```

Comment:
Comments should have the implementation details here - what's the utility to the user of knowing that? The method determines if the cluster is valid.